### DISCUSSION TITLE
`env` options (similar as hyprland)

### DISCUSSION THREAD

**Discussion by Buer-Nahida** [YaLTeR/niri#225] [Ideas]
I think `niri` needs a way to set environment variables in itself config file.

In hyprland, I'm using `env = QT_QPA_PLATFORMTHEME, qt5ct` to set environment variables for window manager.
In niri, I can write `QT_QPA_PLATFORMTHEME=qt5ct` in `/etc/environment`.

But if I do it, these environment variables are play a part in everywhere for my system.
It doesn't make a difference than I set environment variables in window manager config file if I only use wayland.

but I'm not only use niri and other wayland window manager, I also use `bspwm`(x11 window manager), because some x11 apps are doesn't work use xwayland, they are only work in raw x11 environment, If I write `QT_QPA_PLATFORM=wayland` in `/etc/environment`, in `bspwm`, qt apps are all doesn't work.

So `niri` needs a way to set environment variables only for itself. such as this
```kdl
env {
    QT_QPA_PLATFORMTHEME "qt5ct"
}
```

**Comment by YaLTeR** [comment]
Hmm, I see your point. I guess this could be added, not sure if there's a better way.

**Comment by Buer-Nahida** [comment]
Finished
